Automattic/wp-calypso
# Payment Methods: Modularize TaxInfoArea so it can be used in other places and allow editing

#### Changes proposed in this Pull Request

The "Change payment method" page contains a component, `TaxInfoArea`, that was added by https://github.com/Automattic/wp-calypso/pull/57807 to both display the tax location for a saved credit card and to allow editing that information. 

<img width="595" alt="Screen_Shot_2022-04-08_at_10_20_50_AM" src="https://user-images.githubusercontent.com/2036909/162456604-680a503b-fa07-4d73-94a4-db6b6a0ab52e.png">

We'd like to also use this feature on the "Payment methods" page (https://github.com/Automattic/wp-calypso/pull/62576). However, that component relied on a considerable amount of external state, styles, server queries, and dialog components which were interwoven with the saved card payment method label.

This PR moves all needed state, components, styles, and functionality inside `TaxInfoArea` itself so that it can more easily be injected into other places in calypso.

<img width="591" alt="Screen Shot 2022-04-08 at 3 03 39 PM" src="https://user-images.githubusercontent.com/2036909/162507976-765d1c0b-8040-46d9-a79d-6e9dc3e837b6.png">

It also makes three other notable changes:

- When showing the tax location, it now prepends the word "Billing information:". So previously it may have read `10001, US` and now it will read `Billing information: 10001, US`. If the tax location is not set, the edit button will now be prefixed with "Billing information:" also and the button will look more like a link.
- Previously when the tax location was set, it could no longer be edited. This was a deliberate choice in the original implementation due to various UX considerations but I believe I've resolved them here. Now the tax location is clickable and will allow editing.
- The missing tax location info message now reads "Fix missing data" to make it clear that you can (and must) click it.

#### Testing instructions

- Use an account with at least one active subscription.
- Use an account with at least one saved card; ideally one which has no saved tax location information.
- Visit `/me/purchases` and click on an active subscription.
- Click "Change payment method" (or "Add payment method" if no payment method is set).
- Verify that the saved cards display either `Billing information: [POSTAL CODE], [COUNTRY CODE]` or `Fix missing data`.
- Click on the tax location information to bring up the edit dialog.
- Change the country code and postal code in the dialog to be invalid (eg: leave postal code empty) and click "Save".
- Verify that you see an error message about invalid tax location information.
- Change the country code and postal code in the dialog to be valid and click "Save".
- Verify that the displayed tax location now shows the changes.
